lib/realtime/tenants/repo/migrations/20240919140541_reduce_grants_postgres_user.ex

```diff
@@ -0,0 +1,17 @@
+defmodule Realtime.Tenants.Migrations.ReduceGrantsPostgresUser do
+  @moduledoc false
+  use Ecto.Migration
+
+  def change do
+    execute("revoke supabase_realtime_admin from postgres")
+
+    execute("revoke all on table realtime.schema_migrations from postgres")
+    execute("grant select on table realtime.schema_migrations to postgres with grant option")
+
+    execute("revoke all on table realtime.messages from postgres")
+    execute("grant select on table realtime.messages to postgres with grant option")
```

Comment:
actually that is a good call... we might just want to run`insert` as we will have triggers that the users will be able to set against this table and I guess those are ran with the postgres role correct?

if so we want them to also have `insert` but update won't be need and could mess our plans for postgres scaled 🤔 

see https://github.com/supabase/realtime/pull/1156 for more context

Reply:
> triggers that the users will be able to set against this table and I guess those are ran with the postgres role correct?

Yes if the trigger function is `security definer`, otherwise it's whichever role is doing the insert. How are the inserts done?

Reply:
honestly it will depend a lot as they potentially could use authenticated users to achieve it (e.g. postgrest request that inserts something to a table that has a trigger)

Reply:
I see, then we just need to do the same thing we do for Auth triggers

Reply:
got it! thank you 🙏